src/application/controllers/api/v3/Appointmentsv3.php

```diff
@@ -104,23 +109,114 @@ public function getUserAppointments($id_integrated) {
                     ->paginate()
                     ->minimize()
                     ->output();
-            }
-            else {
+            } else {
                 throw new \EA\Engine\Api\V1\Exception('id_user_integrated is required', 400);
             }
-            
+
         } catch (\Exception $exception) {
-                    exit($this->_handleException($exception));
+            exit($this->_handleException($exception));
         }
     }
 
-    private function encodedAppointments($appointments){
-        $encodedAppointments = [];  
-        foreach ($appointments as &$value){
-            array_push($encodedAppointments,$this->parser->customEncode($value));
+    private function encodedAppointments($appointments)
+    {
+        $encodedAppointments = [];
+        foreach ($appointments as &$value) {
+            array_push($encodedAppointments, $this->parser->customEncode($value));
         }
         return $encodedAppointments;
     }
 
-    
+    public function getAppointmentsAddressStatistic($idServiceIntegrated)
+    {
+        try {
+            $requestParams = $this->validateInputRequestParams();
+            $result = [];
+            foreach ($requestParams['cities'] as $cityId) {
+                $element['id'] = $cityId;
+                $element['city'] = $this->cities_model->findCityBy($cityId)[0]['name'];
+                $element['data'] = $this->appointments_model_v3
+                    ->getAddressBookingStatistic($idServiceIntegrated,
+                                                $requestParams['cities'],
+                                                $requestParams['startDate'],
+                                                $requestParams['endDate'],
+                                                $requestParams['gender'],
+                                                $requestParams['firstTime'],
+                                                $requestParams['bhyt'],
+                                                $requestParams['id_provider_integrated']);
+                array_push($result, $element);
+            }
+            $response = new Response($result);
+            $response->output();
+        } catch (\Exception $exception) {
+            exit($this->_handleException($exception));
+        }
+    }
+
+    private function validateInputRequestParams()
+    {
+        $cities = $this->input->get('cities');
+        $startDate = $this->input->get('startDate');
+        $endDate = $this->input->get('endDate');
+        $gender = $this->input->get('gender');
+        $idProviderIntegrated = $this->input->get('id_provider_integrated');
+        $firstTime = $this->checkIsValidBooleanType($this->input->get('firstTime')) ? $this->input->get('firstTime') : null;
```

Comment:
can you check if codeigniter ships with request validation ultil ?
if so please apply it

Reply:
applied